addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,401 @@
+            GregorianCalendar calendar = new GregorianCalendar();
+            calendar.setTime(nextTimer);
+            return new DateTimeType(calendar);
+        } catch (ParseException e) {
+            logger.error("Could not parse date: {}T{} with pattern 'dd.MM.yy'T'HH:mm:ss'", date, time, e);
+            return UnDefType.UNDEF;
+
+        }
+    }
+
+    private void refreshVersionInfo() throws InterruptedException {
+
+        VersionInfo info = robonectClient.getVersionInfo();
+        if (info.isSuccessful()) {
+            updateState(CHANNEL_VERSION_VERSION, new StringType(info.getRobonect().getVersion()));
+            updateState(CHANNEL_VERSION_COMPILED, new StringType(info.getRobonect().getCompiled()));
+            updateState(CHANNEL_VERSION_COMMENT, new StringType(info.getRobonect().getComment()));
+            updateState(CHANNEL_VERSION_SERIAL, new StringType(info.getRobonect().getSerial()));
+        } else {
+            logger.error("Could not retrieve mower version info. Robonect error response message: {}",
+                    info.getErrorMessage());
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+            RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                    robonectConfig.getPassword());
+            httpClient = new HttpClient();
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+            refreshVersionInfo();
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        refreshMowerInfo();
+                        updateStatus(ThingStatus.ONLINE);
+                    } catch (RobonectCommunicationException rce) {
+                        logger.info("Failed to communicate with the mower. Taking it offline.", rce);
+                        updateState(CHANNEL_STATUS, new DecimalType(MowerStatus.OFFLINE.getStatusCode()));
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, rce.getMessage());
+                    } catch (Exception e) {
+                        logger.error("Unexpected exception. Setting thing offline", e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+                    }
+                }
+            };
+            int pollInterval = robonectConfig.getPollInterval() > 0 ? robonectConfig.getPollInterval() : 30;
+            pollingJob = scheduler.scheduleAtFixedRate(runnable, 0, pollInterval, TimeUnit.SECONDS);
+
+            updateStatus(ThingStatus.ONLINE);
+
+        } catch (Exception e) {
+            logger.error("Exception when trying to initialize", e);
+            updateStatus(ThingStatus.UNINITIALIZED);
```

Comment:
If you say so... But why is the ThingStatus defining this value than? If an exception would happen here, indeed the thing would not be initialized properly... So Offline might not be the right state.